Extensions/TileMap/collision/TransformedTileMap.ts

```diff
@@ -0,0 +1,742 @@
+      }
+
+      /**
+       * @returns The tile height in pixels.
+       */
+      getTileWidth() {
+        return this._source.getTileWidth();
+      }
+
+      /**
+       * @returns The number of tile columns in the map.
+       */
+      getDimensionX() {
+        return this._source.getDimensionX();
+      }
+
+      /**
+       * @returns The number of tile rows in the map.
+       */
+      getDimensionY() {
+        return this._source.getDimensionY();
+      }
+
+      /**
+       * @param tileId The tile identifier
+       * @returns The tile definition form the tile set.
+       */
+      getTileDefinition(tileId: integer) {
+        return this._source.getTileDefinition(tileId);
+      }
+
+      /**
+       * @param layerId
+       * @returns The layer identifier.
+       */
+      getLayer(layerId: integer): TransformedCollisionTileMapLayer | undefined {
+        return this._layers.get(layerId);
+      }
+
+      /**
+       * @returns All the layers of the tile map.
+       */
+      getLayers(): Iterable<TransformedCollisionTileMapLayer> {
+        return this._layers.values();
+      }
+
+      /**
+       * Check if a point is inside a tile with a given tag.
+       *
+       * It doesn't use the tile hitboxes.
+       * It only check the point is inside the tile square.
+       *
+       * @param x The X coordinate of the point to check.
+       * @param y The Y coordinate of the point to check.
+       * @param tag The tile tag
+       * @returns true when the point is inside a tile with a given tag.
+       */
+      pointIsInsideTile(x: float, y: float, tag: string): boolean {
+        const workingPoint: FloatPoint =
+          TransformedCollisionTileMap.workingPoint;
+        workingPoint[0] = x;
+        workingPoint[1] = y;
+        this._inverseTransformation.transform(workingPoint, workingPoint);
+        return this._source.pointIsInsideTile(
+          workingPoint[0],
+          workingPoint[1],
+          tag
+        );
+      }
+
+      /**
+       * @param tag The tile tag.
+       * @param left The left border of the area in the scene.
+       * @param top The top border of the area in the scene.
+       * @param right The right border of the area in the scene.
+       * @param bottom The left border of the area in the scene.
+       * @returns At least all the hitboxes from the given area
+       * where tiles have the right tag.
+       *
+       * @see {@link gdjs.RuntimeObject.getHitboxesAround}
+       */
+      getHitboxesAround(
+        tag: string,
+        left: float,
+        top: float,
+        right: float,
+        bottom: float
+      ): Iterable<gdjs.Polygon> {
+        // Return the hitboxes from the tiles that overlap
+        // the AABB of the area in the tile map basis.
+        // Some of these tiles are not event in the given area
+        // but this is a good trade of between the number of
+        // useless returned hitboxes and the time to find them.
```

Comment:
This is what is used for the implementation of `getHitboxesAround`. This is O(1) over the number of tiles in the map.